pyiceberg/table/__init__.py

```diff
@@ -2492,21 +2492,22 @@ def update_column(
                 except ResolveError as e:
                     raise ValidationError(f"Cannot change column type: {full_name}: {field.field_type} -> {field_type}") from e
 
+        # if other updates for the same field exist in one transaction:
         if updated := self._updates.get(field.field_id):
             self._updates[field.field_id] = NestedField(
                 field_id=updated.field_id,
                 name=updated.name,
                 field_type=field_type or updated.field_type,
-                doc=doc or updated.doc,
-                required=updated.required,
+                doc=doc if doc is not None else updated.doc,
+                required=required or updated.required,
```

Comment:
😱 

This looks correct to me. Thank you for the fix!

Reply:
Actually @TiansuYu - I think column requirement setting is handled here: https://github.com/apache/iceberg-python/pull/1083/files#diff-23e8153e0fd497a9212215bd2067068f3b56fa071770c7ef326db3d3d03cee9bR2518

This is because changing the required flag involves checking if incompatible changes are allowed on this `UpdateSchema` instance

Reply:
Hi @TiansuYu please take a look at this comment and let me know what you think!